### ISSUE TITLE
How to use L1 loss for unconditional GAN ?

### ISSUE THREAD

**Issue by ahmed-fau** [junyanz/pytorch-CycleGAN-and-pix2pix#293]
Hi,

According to the paper, L1 seems to be efficient for reconstruction artifact reduction as well as low frequency component estimation. I am thinking about using it for 'unconditional' case !

Would it make sens to 'randomly' select a ground truth data sample to be compared w.r.t the generated fake signal based on L1 loss ? I couldn't get clear idea about how this is implemented for the conditional case here because I don't understand what 'A_real' and 'B_real' refer to.

**Comment by phillipi** [comment]
It's an interesting idea to try an L1 loss in the unconditional case, and related to some popular methods! As you suggest, you could create a randomly paired dataset of {z, image} pairs, and then train min_f |f(z) - image|. However, this may require f to be a very convoluted function, since nearby z's might be forced to map to very different images. So we probably don't want to match z's with completely randomly chosen images. 

Instead you could try to match z's the "right" images according to the current f, e.g., try to find a function g(image) --> z that minimizes the above L1 loss: min_g |f(g(image)) - image|. Putting it all together we get an autoencoder: min_{f,g} |f(g(image)) - image|, and, with a few small modifications to make it probabilistic, we get a VAE, which is an unconditional generative model trained with an L1 (or, more often, L2) loss. You can also add an additional GAN loss, giving a [VAE-GAN](https://arxiv.org/abs/1512.09300?context=cs).

This is just one way to train an unconditional model with an L1 loss between generated samples and 
"randomly" (but intelligently) selected images. There may be some other interesting ways to do it waiting to be discovered. Here is one paper more related paper, which goes the other way around, learning to map images to "random" noise vectors and getting good image features out of the process: https://arxiv.org/pdf/1704.05310.pdf

**Comment by ahmed-fau** [comment]
I think the following paper proposes a solution to that issue:

https://arxiv.org/pdf/1810.06758.pdf

**Comment by John1231983** [comment]
@phillipi : Nice idea. Could you tell me what we expect when adding VAE in the generator? What is the purpose of the adding VAE? Thanks

**Comment by phillipi** [comment]
Here's a paper that talks about the utility of VAE-GAN hybrids: https://arxiv.org/pdf/1706.04987.pdf
